File: client_go/pkg/bomber/logger.go
package bomber

import (
	"fmt"
	"log"
)

const (
	logPrefix = "[BOMBER] "
	// ANSI color codes
	colorReset = "\033[0m"
	colorRed   = "\033[31m"
	colorGreen = "\033[32m"
	colorBlue  = "\033[34m"
	colorCyan  = "\033[36m"
)

func green(format string, v ...any) string {
	message := fmt.Sprintf(format, v...)
	return fmt.Sprintf("%s%s%s", colorGreen, message, colorReset)
}

func blue(format string, v ...any) string {
	message := fmt.Sprintf(format, v...)
	return fmt.Sprintf("%s%s%s", colorBlue, message, colorReset)
}

func red(format string, v ...any) string {
	message := fmt.Sprintf(format, v...)
	return fmt.Sprintf("%s%s%s", colorRed, message, colorReset)
}

// success logs a formatted success message in green
func success(format string, v ...any) {
	message := fmt.Sprintf(format, v...)
	log.Printf("%s%s%s%s\n", logPrefix, colorGreen, message, colorReset)
}

// error logs a formatted error message in red
func error(format string, v ...any) {
	message := fmt.Sprintf(format, v...)
	log.Printf("%s%s%s%s\n", logPrefix, colorRed, message, colorReset)
}

// info logs a formatted info message in blue
func info(format string, v ...any) {
	message := fmt.Sprintf(format, v...)
	log.Printf("%s%s%s%s\n", logPrefix, colorBlue, message, colorReset)
}

// debug logs a formatted debug message in cyan
func debug(format string, v ...any) {
	message := fmt.Sprintf(format, v...)
	log.Printf("%s%s%s%s\n", logPrefix, colorCyan, message, colorReset)
}

File: client_js/pkg/bomber/logger.js
const util = require('util');

const logPrefix = "[BOMBER] ";
const colorReset = "\x1b[0m";
const colorRed = "\x1b[31m";
const colorGreen = "\x1b[32m";
const colorBlue = "\x1b[34m";
const colorCyan = "\x1b[36m";

function green(format, ...args) {
    const message = util.format(format, ...args);
    return `${colorGreen}${message}${colorReset}`;
}

function blue(format, ...args) {
    const message = util.format(format, ...args);
    return `${colorBlue}${message}${colorReset}`;
}

function red(format, ...args) {
    const message = util.format(format, ...args);
    return `${colorRed}${message}${colorReset}`;
}

function success(format, ...args) {
    const message = util.format(format, ...args);
    console.log(`${logPrefix}${colorGreen}${message}${colorReset}`);
}

function error(format, ...args) {
    const message = util.format(format, ...args);
    console.log(`${logPrefix}${colorRed}${message}${colorReset}`);
}

function info(format, ...args) {
    const message = util.format(format, ...args);
    console.log(`${logPrefix}${colorBlue}${message}${colorReset}`);
}

function debug(format, ...args) {
    const message = util.format(format, ...args);
    console.log(`${logPrefix}${colorCyan}${message}${colorReset}`);
}

module.exports = {
    green,
    blue,
    red,
    success,
    error,
    info,
    debug,
};
